k2/python/k2/fsa_algo.py

```diff
@@ -257,3 +258,58 @@ def add_epsilon_self_loops(fsa: Fsa) -> Fsa:
         setattr(out_fsa, name, value)
 
     return out_fsa
+
+
+def remove_epsilon(fsa: Fsa) -> Fsa:
+    '''Remove epsilons (symbol zero) in the input Fsa.
+
+    Caution:
+      It only works on for CPU and doesn't support autograd.
+
+    Args:
+      fsa:
+        The input FSA. It can be either a single FSA or a FsaVec.
+        Must be top-sorted.
+    Returns:
+        The result Fsa, it's equivalent to the input ``fsa`` under
+        tropical semiring but will be epsilon-free.
+        It will be the same as the input ``fsa`` if the input
+        ``fsa`` is epsilon-free. Otherwise, a new epsilon-free fsa
+        is returned and the input ``fsa`` is NOT modified.
+    '''
+    properties = getattr(fsa, 'properties', None)
+    if properties is not None and is_epsilon_free(properties):
+        return fsa
+
+    ragged_arc = _k2.remove_epsilon(fsa.arcs)
+    out_fsa = Fsa.from_ragged_arc(ragged_arc)
```

Comment:
@danpovey, I just realized we may need arc_map here, if not, how can we keep aux_labels after rm-epsilon? we need aux_labels (e.g. words) in decoding, right? I will do this in another PR.